lib/lrama/reporter/conflicts.rb

```diff
@@ -4,30 +4,42 @@
 module Lrama
   class Reporter
     class Conflicts
-      # (IO io, Lrama::States states) -> void
+      # @rbs (IO io, Lrama::States states) -> void
       def report(io, states)
+        if report_conflicts(io, states)
+          io << "\n\n"
+        end
+      end
+
+      private
+
+      # @rbs (IO io, Lrama::States states) -> void
+      def report_conflicts(io, states)
         has_conflict = false
 
         states.states.each do |state|
-          messages = [] #: Array[string]
-          cs = state.conflicts.group_by(&:type)
-          if cs[:shift_reduce]
-            messages << "#{cs[:shift_reduce].count} shift/reduce"
-          end
-
-          if cs[:reduce_reduce]
-            messages << "#{cs[:reduce_reduce].count} reduce/reduce"
-          end
+          messages = format_conflict_messages(state.conflicts)
 
           unless messages.empty?
             has_conflict = true
             io << "State #{state.id} conflicts: #{messages.join(', ')}\n"
           end
         end
 
-        if has_conflict
-          io << "\n\n"
-        end
+        has_conflict
```

Comment:
[nits] I prefer to keep `io << "\n\n"` in `#report_conflicts`.

Reply:
Indeed! revert: https://github.com/ruby/lrama/pull/581/commits/d1eb255f1bfceba851105c95b5d9819b8c190a9e